CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -951,7 +951,7 @@ public static function formRule($fields, $files, $self) {
       return $errors;
     }
 
-    if (CRM_Utils_Array::value('payment_processor_id', $fields) == NULL) {
+    if (CRM_Utils_Array::value('payment_processor_id', $fields) === NULL) {
```

Comment:
the form passes '0' as a string & the test passes 0 as an integer. I fixed to support both as I think it would be easy for it to be cast at some point & both indicate payment processor 0 - ie. pay later